quipucords/scanner/manager.py

```diff
@@ -44,9 +44,9 @@ class Manager(Thread):
     def __init__(self):
         """Initialize the manager."""
         Thread.__init__(self)
-        self.scan_queue = []
-        self.current_job_runner = None
-        self.terminated_job_runner = None
+        self.scan_queue = []  # List[ScanJobRunner]
+        self.current_job_runner = None  # ScanJobRunner
+        self.terminated_job_runner = None  # ScanJobRunner
```

Comment:
mmmm looking at some [examples on pep-484](https://peps.python.org/pep-0484) I'm under the impression this should be preceded by "type" (like `# type: List[ScanJobRunner]`).

I'll give this branch a spin on VSCode to see if it can recognize these bad boys 👀

Reply:
Oh, I wasn't actually using type hints there. I was just typing those out as comments for my human eyes. 😆 

But if I can easily convert them to _real_ type hints, I should.

Reply:
Aha! `# type:` does activate hinting as you said. I may clean this up a bit more.

Reply:
What minimum version of Python do we support in local development again? 😅 

- `# type: ScanJobRunner | None` would be preferred for Python 3.10+
- `# type: Optional[ScanJobRunner]` is what's supported in older Python 3.9.

It probably doesn't really matter since only our IDEs care about them, if they care at all. I may use the modern style here.

Reply:
mmm I know some tests break with python 3.11. not sure about 3.10. I tend to prefer running the same version we are supporting, but (I feel) we are pretty close to update the python version.

but specifically for type hinting we can try adding `from __future__ import annotations`

Reply:
but that's just me, I don't want to impose anything. We already have CI with 3.9 and our image is using it. That should be enough

Reply:
aaaand with `from __future__ import annotations` we can use `list[some_object]`  directly, instead of importing `List` from `typing` module

Reply:
I'll use the 3.9 style for `Optional` for consistency then, and I'll pull `from __future__` like you suggested for `list`.

Reply:
perfect